realm/realm-library/src/androidTest/java/io/realm/RealmQueryTests.java

```diff
@@ -2828,7 +2454,7 @@ public void isValid_removedParent() {
     }
 
     @Test
-    public void isEmpty() {
+    public void isEmpty() throws IOException {
         createIsEmptyDataSet(realm);
         for (RealmFieldType type : SUPPORTED_IS_EMPTY_TYPES) {
```

Comment:
Just curious: Why are we looping and switching instead of just writing the asserts one by one? It would take up 3 times less code?

Reply:
Because we can assert that we have tested all by throwing an exception in the default case. This ensure that the test is failing if we ever introduces new types, etc.

Reply:
Yes - that was my guess. It still looks a bit excessive to do that in every single test here? I imagine perhaps one test would be enough perhaps with a comment to go through all the tests here? But NP, I was just curious if it was anything else.